### PR TITLE
ITE: drivers/espi: fix the VW valid flag issue in MAFS mode

### LINKED ISSUE #52776: ite: eSPI driver: espi_it8xxx2_send_vwire() is not setting valid flag along with respective virtual wire when invoked from app code.
espi_it8xxx2_send_vwire_issue

**Describe the bug**
ite: eSPI driver: espi_it8xxx2_send_vwire() is not setting valid flag along with respective virtual wire when invoked from app code.

When any virtual wire is set/clear from app code using espihub_send_vw() API it is invoking espi_it8xxx2_send_vwire(). But espi_it8xxx2_send_vwire() is only setting respective virtual wire and NOT corresponding valid flag. Even though espi_vw_set_valid() function is available there is no API to invoke it from app code.

**Expected behavior**
espi_it8xxx2_send_vwire() should also set respective valid flag along with virtual wire.

**Impact**
Not able to boot to OS(Windows) in MAFS mode.

### PR DESCRIPTION
**Pull request by bjchentc**
Add CONFIG_ESPI_SET_VW_VALID_WITH_LEVEL option to fix the eSPI driver VW Valid Bits Issue in MAFS mode

fixes #52776

Signed-off-by: BJ Chen <bj.chen@ite.com.tw>

### CODE REVIEW

File: drivers/espi/Kconfig.it8xxx2 [hunk 1]
```diff
@@ -125,4 +125,9 @@ config ESPI_IT8XXX2_PNPCFG_DEVICE_KBC_MOUSE
 	  With this option enabled, EC will send IRQ12 signal to host when the
 	  KBC mouse output buffer is full.
 
+config ESPI_SET_VW_VALID_WITH_LEVEL
```
Comment (keith-zephyr) [inline drivers/espi/Kconfig.it8xxx2:128]: Is this Kconfig needed?  Should the valid bit always be set for supported virtual wires?

I see at least one other [driver](https://github.com/zephyrproject-rtos/zephyr/blob/392a753ac20eb46ba782b5484b2028438f9d9e5e/drivers/espi/espi_npcx.c#L1249) unconditionally sets the valid bit for all virtual wires at init time.

File: drivers/espi/espi_it8xxx2.c [hunk 2]
```diff
@@ -696,11 +696,18 @@ static int espi_it8xxx2_send_vwire(const struct device *dev,
...
+#ifdef CONFIG_ESPI_SET_VW_VALID_WITH_LEVEL
+	uint8_t valid_mask = vw_channel_list[signal].valid_mask;
+#endif
 
 	if (signal > ARRAY_SIZE(vw_channel_list)) {
 		return -EIO;
 	}
 
+#ifdef CONFIG_ESPI_SET_VW_VALID_WITH_LEVEL
+	vw_reg->VW_INDEX[vw_index] |= valid_mask;
```
Comment (keith-zephyr) [inline drivers/espi/espi_it8xxx2.c:708]: Use `espi_vw_set_valid()` instead.  Also consider whether this can be set once at initialization time.

**Review by keith-zephyr — CHANGES_REQUESTED** [review]

File: drivers/espi/Kconfig.it8xxx2 (hunk 1 above)
Comment (itebj) [reply drivers/espi/Kconfig.it8xxx2:128]: Yes, sir. 
Indeed, it should be set for all the virtual wires, I'll removed the Kconfig option.

File: drivers/espi/espi_it8xxx2.c (hunk 2 above)
Comment (itebj) [reply drivers/espi/espi_it8xxx2.c]: Dear sir,

We've modified it again to remove the espi_vw_set_valid() and put the valid bits settings together with the espi_it8xxx2_send_vwire(). Meanwhile, the ESPI_SET_VW_VALID_WITH_LEVEL has been removed as well.

Thanks for guidance & regards,
BJ

**Review by keith-zephyr — APPROVED** [review]

**Review by Dino-Li — APPROVED** [review]

**Pull request merged**